lightning/src/offers/offer.rs

```diff
@@ -8,19 +8,212 @@
+			quantity_max: None, node_id: Some(node_id),
+		};
+		OfferBuilder { offer }
+	}
+
+	/// Sets a chain hash of the given [`Network`] for the offer. If not called,
+	/// [`Network::Bitcoin`] is assumed.
+	///
+	/// Successive calls to this method will add another chain hash.
+	pub fn chain(mut self, network: Network) -> Self {
+		let chains = self.offer.chains.get_or_insert_with(Vec::new);
+		let chain = ChainHash::using_genesis_block(network);
+		if !chains.contains(&chain) {
+			chains.push(chain);
+		}
+
+		self
+	}
+
+	/// Sets the metadata for the offer. Useful for authentication and validating fields.
+	///
+	/// Successive calls to this method will override the previous setting.
+	pub fn metadata(mut self, metadata: Vec<u8>) -> Self {
+		self.offer.metadata = Some(metadata);
+		self
+	}
+
+	/// Sets the amount for the offer.
+	///
+	/// Successive calls to this method will override the previous setting.
+	pub fn amount(mut self, amount: Amount) -> Self {
+		self.offer.amount = Some(amount);
+		self
+	}
+
+	/// Sets the features for the offer.
+	///
+	/// Successive calls to this method will override the previous setting.
+	#[cfg(test)]
+	pub fn features(mut self, features: OfferFeatures) -> Self {
+		self.offer.features = Some(features);
+		self
+	}
+
+	/// Sets the absolute expiry for the offer as seconds since the Unix epoch.
+	///
+	/// Successive calls to this method will override the previous setting.
+	pub fn absolute_expiry(mut self, absolute_expiry: Duration) -> Self {
+		self.offer.absolute_expiry = Some(absolute_expiry);
+		self
+	}
+
+	/// Sets the issuer for the offer.
+	///
+	/// Successive calls to this method will override the previous setting.
+	pub fn issuer(mut self, issuer: String) -> Self {
+		self.offer.issuer = Some(issuer);
+		self
+	}
+
+	/// Sets a blinded path for the offer.
+	///
+	/// Successive calls to this method will add another blinded path.
+	pub fn path(mut self, path: BlindedPath) -> Self {
+		self.offer.paths.get_or_insert_with(Vec::new).push(path);
+		self
+	}
+
+	/// Sets a fixed quantity of items for the offer. If not set, `1` is assumed.
+	///
+	/// Successive calls to this method or [`quantity_range`] will override the previous setting.
+	///
+	/// [`quantity_range`]: Self::quantity_range
+	pub fn quantity_fixed(mut self, quantity: NonZeroU64) -> Self {
+		let quantity = Some(quantity.get()).filter(|quantity| *quantity != 1);
+		self.offer.quantity_min = quantity;
+		self.offer.quantity_max = quantity;
+		self
+	}
+
+	/// Sets a quantity range of items for the offer. If not set, `1` is assumed.
+	///
+	/// Successive calls to this method or [`quantity_fixed`] will override the previous setting.
+	///
+	/// [`quantity_fixed`]: Self::quantity_fixed
+	pub fn quantity_range<R: RangeBounds<NonZeroU64>>(mut self, quantity: R) -> Self {
+		self.offer.quantity_min = match quantity.start_bound() {
+			Bound::Included(n) => Some(n.get()),
+			Bound::Excluded(_) => unreachable!(),
```

Comment:
Is this technically not unreachable if the user supplies a custom `RangeBounds` impl?

Reply:
Panic sgtm!